### DISCUSSION TITLE
API routes with Apollo client example in TypeScript

### DISCUSSION THREAD

**Discussion by avxkim** [vercel/next.js#12892] [Help]
https://github.com/zeit/next.js/blob/canary/examples/api-routes-apollo-server-and-client/apollo/client.js:

```
const createIsomorphLink = () => {
  if (typeof window === 'undefined') {
    const { SchemaLink } = require('apollo-link-schema')
    const { schema } = require('./schema')
    return new SchemaLink({ schema })
  } else {
    const { HttpLink } = require('apollo-link-http')
    return new HttpLink({
      uri: '/api/graphql',
      credentials: 'same-origin',
    })
  }
}
```

There's a few problems:

1. What type to return in `createIsomorphLink`
2. How to dynamically import in TypeScript? `const { SchemaLink } = require('apollo-link-schema')`. I've tried `const { SchemaLink } = import('apollo-link-schema');`, but it doesn't work.

**Comment by dan-turner** [comment]
Did you ever figure this one out? Coming across this myself now.
